### ISSUE TITLE
GooglePay environment is not set correctly

### ISSUE THREAD

**Issue by BenedictP** [Adyen/adyen-android#522]
**Describe the bug**
When using the Builder constructor `Builder(@NonNull Context context, @NonNull String clientKey)` inside the `GooglePayConfiguration` class. The adyen test environment is set in the super call resulting that the `getDefaultGooglePayEnvironment()` method is returning the Google Pay test environment. But I'm setting the adyen prod environment later in the builder and this has no impact on the Google Pay environment because the method is already called.

**To Reproduce**
When I create the Google Pay environment like this:
`GooglePayConfiguration.Builder(context, "clientKey").setEnvironment(Environment.LIVE).build()`
The result is that Google Pay is not using the live environment.

But would I use:
`GooglePayConfiguration.Builder(LocaleUtil.getLocale(context), Environment.LIVE, "clientKey").build()`
It would be correct.

**Expected behavior**
I would expect that Google Pay also uses the live environment and it shouldn't matter if I set the environment in the Builder constructor or later with a setEnvironment().

**Comment by jreij** [comment]
Hi @BenedictP , thanks for opening this, interesting thought indeed!

With the current implementation, you could either call:
```kotlin
GooglePayConfiguration.Builder(locale, Environment.LIVE, clientKey).build()
```
or
```kotlin
GooglePayConfiguration.Builder(context, clientKey)
    .setShopperLocale(locale)
    .setEnvironment(Environment.LIVE)
    .setGooglePayEnvironment(WalletConstants.ENVIRONMENT_PRODUCTION)
    .build()
```

We could make sure that calling one of `setEnvironment` or `setGooglePayEnvironment` automatically adjusts the other one (or we could merge both methods into one), but I am wondering if there could be any scenario where someone would want to use Adyen's test environment with GooglePay's live one or vice versa. If that is the case, we would be disabling a valid scenario.

**Comment by jreij** [comment]
Right, I agree that it's not clear. We'll see how it can be improved 👍

**Comment by jreij** [comment]
Issue should be fixed by the linked PR. The new behavior is that whenever you set the Adyen environment by either using the constructor or the `setEnvironment` method, the GooglePay environment will match it automatically, unless you explicitly call `setGooglePayEnvironment`. This exception is added just to provide a way to test with non-matching environments, in case anyone needs that.
